### PR TITLE
Optionally make x,y,z,t available in ParsedAux expressions

### LINKED ISSUE #15877: Optionally make x,y,z,t available in ParsedAux
## Reason
This has been requested before and is easy to implement. There are legitimate reasons to make current simulation time and QP coordinates available in `ParsedAux` for example to blend field variables as a function of time or space.

## Design
Add boolean switch to make x,y,z,t available in the parsed expression. For backwards compatibility this defaults to false (in case field variables with the same names are already being used in existing input files)

## Impact
Added capability.

### PR DESCRIPTION
**Pull request by dschwen**
Refs #15877

### CODE REVIEW

File: framework/src/auxkernels/ParsedAux.C [hunk 1]
```diff
@@ -66,15 +79,31 @@ ParsedAux::ParsedAux(const InputParameters & parameters)
   if (_enable_jit)
     _func_F->JITCompile();
 
+  // obtain current time
+  if (_use_xyzt)
+  {
+    auto fpb = dynamic_cast<FEProblemBase *>(&_subproblem);
+    if (!fpb)
+      paramError("use_xyzt", "Cannot access simulation time t. Set 'use_xyzt' to false.");
+    _time = &fpb->time();
```
Comment (dschwen) [inline framework/src/auxkernels/ParsedAux.C:88]: I need to jump through this little hoop to get access to time. What cases are there that could trigger the error. And would it be better here to just set the time variable to zero and let users carry on?

**Comment by WilkAndy** [comment]
I'd prefer to see use_xyzt defaulting to true and for you to change the tests where it causes failure.

a

**Comment by dschwen** [comment]
@WilkAndy hmmm, the tests are something we control and can change easily. I really wouldn't feel good about breaking other people's input files though. I guess I could check if x,y,z, or t are used as field variables and then not pull those in from coordinates or time, but I think this would just make things confusing AF.

**Comment by WilkAndy** [comment]
> @WilkAndy hmmm, the tests are something we control and can change easily. I really wouldn't feel good about breaking other people's input files though. I guess I could check if x,y,z, or t are used as field variables and then not pull those in from coordinates or time, but I think this would just make things confusing AF.

I understand, but you could say the same thing about many additions to MOOSE.  Perhaps you could include a warning if you detect "x", etc, as field variables?

**Comment by dschwen** [comment]
Ok, I really need this to be in. I understand Andy's desire to have this enabled by default, but I'd rather revisit that point in a separate issue. Such a change would be more invasive than this PRs and I prefer a gentler approach here.

**Review by loganharbour — APPROVED** [review]

**Comment by tophmatthews** [comment]
How about parsedmaterials?

**Comment by tophmatthews** [comment]
good point, I was mostly thinking about `time` to avoid `TimeStepMaterial`

**Pull request merged**